src/lib/geo-location.js

```diff
@@ -40,10 +40,11 @@ const mockLocation = {
  */
 const getGeoLocation = async ({ geoCountry, mode, offline, state }) => {
   const cacheObject = state.get(STATE_GEO_PROPERTY)
+  if (geoCountry) mode = 'mock'
 
   // If we have cached geolocation data and the `--geo` option is set to
   // `cache`, let's try to use it.
-  if (cacheObject !== undefined && mode === 'cache') {
+  if (cacheObject !== undefined && (mode === 'cache' || cacheObject.data.country.code === geoCountry)) {
```

Comment:
```suggestion
  // Or, if the country we're trying to mock is the same one as we have in the
  // cache, let's use the cache instead of the mock.
  if (cacheObject !== undefined && (mode === 'cache' || cacheObject.data.country.code === geoCountry)) {
```